src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ToDoCommand;
+import duke.exception.DukeInputException;
+
+/** Contains functions for parsing user inputs to software commands */
+public class Parser {
+
+    /**
+     * Parses given input String into commands with parameters.
+     *
+     * @param userCommand Input string, typically from user.
+     * @return Command object that represents the user command and contains execution functionality.
+     * @throws DukeInputException Thrown when input cannot be cleanly parsed into an available command.
+     */
+    public static Command parse(String userCommand) throws DukeInputException {
+
+        // Separates user input into command and parameters
+        String[] inputs = userCommand.split(" ", 2);
+        String command = inputs[0];
+        String params = "";
```

Comment:
I understand why u named it as params, but for the purposes of the style guide could u rename to a singular form?Since plural form namings are reserved for collections.

Reply:
Renamed to `additionalInformation`.